packages/button/components/ButtonBase.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react";
+import { cx } from "emotion";
+import {
+  buttonReset,
+  textWeight,
+  flex,
+  flexItem,
+  display,
+  padding
+} from "../../shared/styles/styleUtils";
+import {
+  button,
+  buttonBase,
+  buttonContent,
+  buttonInverse,
+  fullWidthButton,
+  getInverseMutedButtonStyles,
+  getMutedButtonStyles,
+  processingText
+} from "../style";
+
+export enum ButtonAppearances {
+  Primary = "primary",
+  Secondary = "secondary",
+  Standard = "standard",
+  Danger = "danger",
+  Success = "success"
+}
+
+export interface ButtonProps {
+  /**
+   * if the button triggers new content to appear (e.g.: modals and dropdowns)
+   */
+  ariaHaspopup?: boolean;
+  children?: React.ReactNode | string;
+  /**
+   * whether or not the button is enabled
+   */
+  disabled?: boolean;
+  /**
+   * the icon that appears before the button text
+   */
+  iconStart?: React.ReactElement<HTMLElement>;
+  /**
+   * the icon that appears after the button text
+   */
+  iconEnd?: React.ReactElement<HTMLElement>;
+  /**
+   * whether the action the button was meant to do has completed
+   */
+  isProcessing?: boolean;
+  /**
+   * if the button should fill the width of it's container
+   */
+  isFullWidth?: boolean;
+  /**
+   * if the button is on a dark background
+   */
+  isInverse?: boolean;
+  /**
+   * the function the button calls
+   */
+  onClick?: (e?: React.SyntheticEvent<HTMLElement>) => void;
+  /**
+   * the type of button - same as HTML "type" attribute on the <button> tag
+   */
+  type?: "button" | "reset" | "submit";
+}
+
+export interface ButtonBaseProps extends ButtonProps {
+  appearance: ButtonAppearances;
+  className?: string;
+}
+
+class ButtonBase extends React.PureComponent<ButtonBaseProps, {}> {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+  public render() {
+    const {
+      appearance,
+      children,
+      className,
+      disabled,
+      iconStart,
+      iconEnd,
+      isInverse,
+      isProcessing,
+      isFullWidth,
+      onClick,
+      ...other
+    } = this.props;
+
+    return (
+      <button
+        className={cx(buttonReset, className, {
+          [fullWidthButton]: isFullWidth
+        })}
+        disabled={disabled || isProcessing}
+        onClick={this.onClick}
+        tabIndex={0}
+        {...other}
+      >
+        <span
```

Comment:
This span wrapper might seem like an odd choice, but it's necessary for only showing the `:focus` style to keyboard users.

If anybody is super upset with this markup, I guess I could handle that in JS instead :/